src/gui/plugins/scene_manager/GzSceneManager.cc

```diff
@@ -134,39 +134,42 @@ void GzSceneManager::Update(const UpdateInfo &_info,
   this->dataPtr->renderUtil.UpdateFromECM(_info, _ecm);
 
   // load visual plugin on gui side
-  std::map<Entity, sdf::ElementPtr> pluginElems;
+  std::map<Entity, sdf::Plugins> plugins;
   if (!this->dataPtr->initializedVisualPlugins)
   {
-    _ecm.Each<components::Visual, components::VisualPlugin>(
+    _ecm.Each<components::Visual, components::VisualPlugins>(
         [&](const Entity &_entity,
             const components::Visual *,
-            const components::VisualPlugin *_plugin)->bool
+            const components::VisualPlugins *_plugins)->bool
     {
-      sdf::ElementPtr pluginElem = _plugin->Data();
-      pluginElems[_entity] = _plugin->Data();
+      plugins[_entity].insert(plugins[_entity].end(),
+          _plugins->Data().begin(), _plugins->Data().end());
       return true;
     });
     this->dataPtr->initializedVisualPlugins = true;
   }
   else
   {
-    _ecm.EachNew<components::Visual, components::VisualPlugin>(
+    _ecm.EachNew<components::Visual, components::VisualPlugins>(
         [&](const Entity &_entity,
             const components::Visual *,
-            const components::VisualPlugin *_plugin)->bool
+            const components::VisualPlugins *_plugins)->bool
     {
-      sdf::ElementPtr pluginElem = _plugin->Data();
-      pluginElems[_entity] = _plugin->Data();
+      plugins[_entity].insert(plugins[_entity].end(),
+          _plugins->Data().begin(), _plugins->Data().end());
       return true;
     });
   }
-  for (const auto &it : pluginElems)
+  for (const auto &it : plugins)
   {
-    ignition::gazebo::gui::events::VisualPlugin visualPluginEvent(
-        it.first, it.second);
```

Comment:
We should still send this event for backwards compatibility

Reply:
[61fcf05](https://github.com/gazebosim/gz-sim/pull/1352/commits/61fcf059bdda7db75e59e12058eec67f713d5fed)